drivers/gpio/Kconfig.pcal64xxa

```diff
@@ -18,3 +18,12 @@ config GPIO_PCAL64XXA_INIT_PRIORITY
 	depends on GPIO_PCAL64XXA
 	help
 	  Device driver initialization priority.
+
+config GPIO_PCAL64XXA_RETENTION
```

Comment:
I would let the driver handle this automatically using the devicetree, specifically the macro `DT_ANY_INST_HAS_PROP_STATUS_OKAY` which can be used to check if any node of the gpio controller has optional the property `retention` :)

Reply:
```
#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(retention) && defined(CONFIG_RETENTION)

#endif
```

Reply:
I like the idea, done.